e2e/machineset_migration_test.go

```diff
@@ -213,13 +214,676 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
+						if err != nil {
+							return fmt.Errorf("failed to get CAPI Machines: %w", err)
+						}
+						if len(capiMachines) == 0 {
+							return fmt.Errorf("CAPI Machines were deleted")
+						}
+
+						mapiMachine, err :=mapiframework.GetMachine(cl,capiMachines[0].Name)
+						if err != nil {
+							return fmt.Errorf("failed to get MAPI Machines: %w", err)
+						}
+						if mapiMachine == nil {
+							return fmt.Errorf("MAPI Machine were deleted")
+						}
+
+						return nil
+					}, capiframework.WaitLong, capiframework.RetryLong).Should(Succeed(), "Both MAPI and CAPI Machines should persist for 15 minutes")
+
+					By("Verifying no owner references on MAPI Machines")
+					mapiMachines, err := mapiframework.GetMachinesFromMachineSet(ctx, cl, mapiMachineSet)
+					Expect(err).ToNot(HaveOccurred(), "failed to get MAPI Machines from MachineSet")
+					for _, machine := range mapiMachines {
+						Expect(machine.GetOwnerReferences()).To(BeEmpty(),
+							"MAPI Machine %s should have no owner references", machine.Name)
+					}
+				*/
+			})
+		})
+	})
+
+	var _ = Describe("Update MachineSets", Ordered, func() {
+		var mapiMSAuthMAPIName = "ms-authoritativeapi-mapi"
+		var mapiMachineSet *mapiv1beta1.MachineSet
+		var capiMachineSet *capiv1beta1.MachineSet
+		var awsMachineTemplate *capav1.AWSMachineTemplate
+		var changedAWSMachineTemplate *capav1.AWSMachineTemplate
+		var err error
+
+		BeforeAll(func() {
+			mapiMachineSet, err = createMAPIMachineSetWithAuthoritativeAPI(ctx, cl, 0, mapiMSAuthMAPIName, mapiv1beta1.MachineAuthorityMachineAPI, mapiv1beta1.MachineAuthorityMachineAPI)
+			Expect(err).ToNot(HaveOccurred(), "MAPI MachineSet creation should succeed")
+
+			capiMachineSet, awsMachineTemplate = verifyMAPIMachineSetHasCAPIMirror(cl, mapiMSAuthMAPIName)
+
+			DeferCleanup(func() {
+				By("Cleaning up 'Update MachineSet' resources")
+				cleanupTestResources(
+					ctx,
+					cl,
+					[]*capiv1beta1.MachineSet{capiMachineSet},
+					[]*capav1.AWSMachineTemplate{awsMachineTemplate, changedAWSMachineTemplate},
+					[]*mapiv1beta1.MachineSet{mapiMachineSet},
+				)
+			})
+		})
+
+		Context("when MAPI MachineSet with spec.authoritativeAPI: MAPI and replicas 0", Ordered, func() {
+			It("should be rejected when scaling CAPI mirror", func() {
+				By("Scaling up CAPI MachineSet to 1 should be rejected")
+				capiframework.ScaleMachineSet(mapiMSAuthMAPIName, 1, capiframework.CAPINamespace)
+				mapiMachineSet, err := capiframework.GetMachineSet(cl, mapiMSAuthMAPIName, capiframework.CAPINamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to get mapiMachineSet %s", mapiMachineSet)
+				Eventually(komega.Object(mapiMachineSet)).Should(HaveField("Spec.Replicas", HaveValue(Equal(int32(0)))))
+			})
+
+			It("should be rejected when updating CAPI mirror spec", func() {
+				By("Updating CAPI mirror spec (such as DeletePolicy)")
+				Eventually(func() error {
```

Comment:
Does the get need to be in the `Eventually` wrapper here? 

IIUC `k.Update` wraps an additional get, so we could move it outside and avoid redoing the get if it fails.

Reply:
@sunzhaohua2 nice job on using `k.Update`, but I don't think we need the outer `Eventually()`. :)